direct/core/src/main/java/cz/o2/proxima/direct/batch/BatchLogReaders.java

```diff
@@ -84,6 +89,13 @@ public String toString() {
           .toString();
     }
 
+    @Override
+    public Factory<?> asFactory() {
+      final Factory<?> superFactory = super.asFactory();
+      final ThroughputLimiter limiter = this.limiter;
```

Comment:
OT: this brings up idea whether we need some kind of closureCleaner as flink / spark has... 🤔 Or at least some kind of guard from serializaing large scopes.

Reply:
The guard should be, that we don't have Serializable interface implemented on classes that are not supposed to be serialized. Unfortunately, this is not the case currently.